### PR TITLE
CLI for working with pangeo-forge.

### PR DESCRIPTION
**Pull request by andersy005**
This PR tries to address #43 by building on @TomAugspurger's work in https://github.com/pangeo-forge/pangeo-forge/blob/00306b38addaafa475c9f932c8eec58d4ffa047a/pangeo_forge/cli.py. The implementation is based on @charlesbluca's ideas in https://github.com/pangeo-forge/pangeo-forge/issues/43#issuecomment-767086077. 

So far, there are plenty of placeholders but I could still use some feedback and/or input on which commands and subcommands are needed for the CLI. 

```console
❯ pangeo                                                                                
Usage: pangeo [OPTIONS] COMMAND [ARGS]...

  CLI for working with pangeo-forge.

Options:
  --version             Display pangeo-forge version.
  --install-completion  Install completion for the current shell.
  --show-completion     Show completion for the current shell, to copy it or
                        customize the installation.

  --help                Show this message and exit.

Commands:
  bakery  Managing submissions of recipes to bakeries.
  forge   Running and validating recipes locally.
  smithy  Generation of recipes and validation of YAML configs.
```

```console
❯ pangeo forge                                                                          
Usage: pangeo forge [OPTIONS] COMMAND [ARGS]...

  Running and validating recipes locally.

Options:
  --help  Show this message and exit.

Commands:
  check     Check that the pipeline definition is valid.
  register  Register a pipeline with prefect.
  run       Run a pipeline with prefect.
```

```console
❯ pangeo smithy                                                                          
Usage: pangeo smithy [OPTIONS] COMMAND [ARGS]...

  Generation of recipes and validation of YAML configs.

Options:
  --help  Show this message and exit.

Commands:
  init  Create a new pangeo-forge recipe.
  lint  Lint a single pangeo-forge recipe.

```

```console
❯ pangeo bakery                                                                          
Usage: pangeo bakery [OPTIONS] COMMAND [ARGS]...

  Managing submissions of recipes to bakeries.

Options:
  --help  Show this message and exit.

Commands:
  check
  list
  submit
```

### CODE REVIEW

File: pangeo_forge/cli/forge.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            flow.validate()
+
+    if errors:
+        table = Table()
+        table.add_column("Errors", style="magenta")
+        for error in errors:
+            table.add_row(error)
+
+        console.print(table)
+
```
Comment (TomAugspurger) [inline pangeo_forge/cli/forge.py:55]: Can you make sure this exits with a non-zero exit code, to indicate an error?

```suggestion
        console.print(table)
        sys.exit(1)
```

File: pangeo_forge/cli/forge.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        if "Pipeline" not in result:
+            errors.append(f"File '{recipe}/pipeline.py' must have a class named 'Pipeline'")
+        if "flow" not in result:
+            errors.append(f"File '{recipe}/pipeline.py' must have a prefect Flow named 'flow'")
+        elif not isinstance(result["flow"], prefect.Flow):
+            errors.append(f"File '{recipe}/pipeline.py' must have a prefect Flow named 'flow'")
+        else:
+            flow = result["flow"]
+            flow.validate()
```
Comment (TomAugspurger) [inline pangeo_forge/cli/forge.py:46]: It's worth checking exactly what the new recipe structure requires. Previously we needed things to be valid prefect Flows. Now I'm guessing we'd want to check the types of the various stages.

**Comment by rabernat** [comment]
Thanks so much for this @andersy005!

Going forward, we don't want the user to have to include any of these prefect details in their recipe. So it seems that before we can really move forward with the CLI, we may need to first answer the question of _what is the structure of a recipe repo?_ We need an issue to track that.

**Comment by rabernat** [comment]
See https://github.com/pangeo-forge/pangeo-forge/pull/71 for some discussion of what the recipe contribution process might look like?

**Comment by charlesbluca** [comment]
This looks great! I'm glad my comment was of some use for this.

In terms of (sub)commands, I'm wondering if we would need some form of authentication handling for submission to bakeries - this could be handled either with a subcommand within `pangeo bakery`, or as a command of its own if we expect authentication to be an issue for the `forge` or `smithy` commands.

**Pull request closed without merging**